orb-ui/src/engine/animations/simple_spinner.rs

```diff
@@ -0,0 +1,218 @@
+            transition: None,
+            transition_time: 0.0,
+            transition_background: None,
+        }
+    }
+
+    /// Set the speed of the spinner in radians per second.
+    #[allow(dead_code)]
+    pub fn speed(self, speed: f64) -> Self {
+        Self { speed, ..self }
+    }
+
+    /// Get the current phase of the spinner.
+    /// The phase is the current position of the spinner in radians.
+    pub fn phase(&self) -> f64 {
+        self.phase
+    }
+
+    #[allow(dead_code)]
+    pub fn fade_in(self, duration: f64) -> Self {
+        Self {
+            transition: Some(Transition::FadeIn(duration)),
+            transition_time: 0.0,
+            ..self
+        }
+    }
+
+    pub fn background(&self) -> Argb {
+        self.background
+    }
+}
+
+impl<const N: usize> Animation for SimpleSpinner<N> {
+    type Frame = RingFrame<N>;
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    #[allow(clippy::cast_precision_loss, clippy::float_cmp)]
+    fn animate(
+        &mut self,
+        frame: &mut RingFrame<N>,
+        dt: f64,
+        idle: bool,
+    ) -> AnimationState {
+        if let Some(Transition::ForceStop) = self.transition {
+            return AnimationState::Finished;
+        }
+
+        if let Some(Transition::StartDelay(duration)) = self.transition {
+            self.transition_time += dt;
+            if self.transition_time >= duration {
+                self.transition = None;
+            }
+            return AnimationState::Running;
+        }
+
+        self.phase = (self.phase + dt * self.speed) % (2.0 * PI);
+        // `N - led-index` with `led-index` increasing in {0,N}, because the LED strip goes anti-clockwise
+        // `N as f64 * 3.0 / 4.0` because the first LED is at 6 o'clock (3PI/2)
+        let progress = (N as f64 - (self.phase * N as f64 / (2.0 * PI))
+            + N as f64 * 3.0 / 4.0)
+            % N as f64;
+        let led_index = progress as usize;
+        let head_tail_scale = progress - led_index as f64;
+
+        if !idle {
+            let mut factor = match self.transition {
+                Some(Transition::FadeOut(duration)) => {
+                    self.transition_time += dt;
+                    if self.transition_time >= duration {
+                        return AnimationState::Finished;
+                    }
+                    (self.transition_time * PI / 2.0 / duration).cos()
+                }
+                Some(Transition::FadeIn(duration)) => {
+                    self.transition_time += dt;
+                    if self.transition_time >= duration {
+                        self.transition = None;
+                        self.transition_background = None;
+                    }
+                    (self.transition_time * PI / 2.0 / duration).sin()
+                }
+                _ => 1.0,
+            };
```

Comment:
Consider: Possibly making into a helper function and writing a small unit test, which serves as examples of expected inputs/outputs.